src/odemis/driver/test/tfsbc_test.py

```diff
@@ -60,6 +67,23 @@ def setUpClass(cls):
     def tearDownClass(cls):
         pass
 
+    def test_dependency_scanner(self):
+        """Test dependency on scanner."""
+        # TODO only works with xt simulator now - how to make generic? for xt simulator has special flag "sim"
+        if TEST_NOHW:
```

Comment:
How to handle this correctly. For xt there are 3 options for TEST_HW: 
0 (HW testing), "sim" (simulator testing), 1 (skip most tests). Here we only have 2 options. So if TEST_HW = 1, this test is run, but is then the xt adapter simulator not running?

Reply:
I think that the problem is you're relying on the xt_client connected to xtlib adpater (running in simulator mode on the same computer) to get a component which must now have a beamShiftTransformationMatrix VA.  It's a very big hammer for a tiny nail. I'd suggest to either create a mock component "manually" with the given VA (and you can then pass the right config!), or use a `simulated.GenericComponent`. You can have a fake "e-beam" in 2 lines!

Reply:
I used now the simulated.GenericComponent. Then I can use it directly in the setUpClass for all tests instead of manually adding the matrix to the MD. However, then for all tests the dependency is used. Is it ok to do this? Then we would not need the separate class and all tests pass as usual (I used the hard coded matrix for the simulated component).